### PR TITLE
Issue #7662: Update doc for SuperFinalize 

### PR DESCRIPTION
**Pull request by Shaman20**
Fixes Issue https://github.com/checkstyle/checkstyle/issues/7662
image

Output of default example:
image

### CODE REVIEW

**Comment by Shaman20** [comment]
I am getting a commit error
image

Also I got this message
image

Please help me to resolve this error.

**Comment by romani** [comment]
As mentioned in error messages, you have to:
Fix commit message.
Rebase on latest code.

**Comment by romani** [comment]
Squash commits in one.
Read https://checkstyle.org/beginning_development.html#Starting_Development

**Comment by Shaman20** [comment]
"Updated doc for superfinalize" is my commit message but I got an error
image

**Comment by romani** [comment]
Please read error and do what it demands from you

**Comment by Shaman20** [comment]
> " Updated doc for superFinalize" is my commit message but I got an error image

Hey, tried changing the message as per the pattern given but could not resolve the error still. I don't know what to do. Please help. mvn clean verify runs properly in master branch but it is not running in the new branch I have created which is my-new-check the above error is appearing

**Comment by romani** [comment]
Make commit to start with `Issue #7631: `

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java [hunk 1]
```diff
@@ -39,6 +39,21 @@
...
+ *            System.out.println("Finalize of Sub Class"); // Release resources, perform cleanup
+ *        } catch(Throwable t){
+ *            throw t;
+ *        } finally{
+ *            System.out.println("Calling finalize of Super Class");
+ *            super.finalize();
+ *        }
+ *     }
+ * }
+ * </pre>
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java:56]: Please add one more example with violation. To let people see not good example

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java (hunk 1 above)
Comment (Shaman20) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java:56]: Yeah sure

Comment (Shaman20) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java:69]: image
image

The number of characters and spaces used are exact in both config_coding.xml and SuperFinalizeCheck.java file but I am unable to resolve the error 
- Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
        at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.examineCheckSection(XdocsJavaDocsTest.java:191)
        at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.testAllCheckSectionJavaDocs(XdocsJavaDocsTest.java:164)

Also I'm unable to find as the above displayed path is not found on my machine "XdocsJavaDocsTest.java"
Please help

File: src/xdocs/config_coding.xml [hunk 2]
```diff
@@ -7088,6 +7088,35 @@ class C {
...
+}
+class B {
+   @Override
+      protected void finalize() throws Throwable {
+            try{
+                // Statements;
+            } catch() {
+                //body;
+            }
+            finally { // causes violation does not call super.finalize
```
Comment (romani) [inline src/xdocs/config_coding.xml:7114]: Please put comment "//violation" on line where checkstyle put violation.
Prove it CLI execution.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java [hunk 3]
```diff
@@ -39,6 +39,34 @@
  * <pre>
  * &lt;module name=&quot;SuperFinalize&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class A {
+ *  @Override
+ *      protected void finalize throws Throwable {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java:46]: `()` is missing.
Please make sure code is compilable

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java:69]: First image shows that sources are different, looks attentively to spot a diff

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java (hunk 3 above)
Comment (Shaman20) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java:46]: Sure, I'll do it. Thank you so much

File: src/xdocs/config_coding.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/config_coding.xml]: Please reply "done" to each review item you addressed and pushed a fix

**Comment by romani** [comment]
Use this, it is more compact
```
@Override
  protected void finalize() throws Throwable {
    System.out.println("In finalize block");
    super.finalize();
  }
```

Please show CLI execution on your examples. Put violation comment on line where CLI gives you violation

File: src/xdocs/config_coding.xml [hunk 4]
```diff
@@ -7088,6 +7088,21 @@ class C {
         <source>
 &lt;module name=&quot;SuperFinalize&quot;/&gt;
         </source>
+        <p>Example:</p>
+        <source>
+public class A {
+      protected void finalize() throws Throwable {
+            System.out.println("In finalize block");
+            super.finalize(); //OK, calls super.finalize()
```
[the remark this answers is not in the document]
Comment (Shaman20) [reply src/xdocs/config_coding.xml:7096]: done

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by romani** [comment]
@Shaman20, thanks a lot for persistence to make it happen.

Please do not resolve my review items, just reply "done".

Please fix one more documentation-examples issue to make sure you firm in your knowledge and can do it without bunch to-and-from in OR comments.

**Comment by Shaman20** [comment]
> @Shaman20, thanks a lot for persistence to make it happen.
> 
> Please do not resolve my review items, just reply "done".
> 
> Please fix one more documentation-examples issue to make sure you firm in your knowledge and can do it without bunch to-and-from in OR comments.

Thank you so much for being cooperative and had a great time contributing

**Pull request merged**